minidns-dnssec/src/main/java/de/measite/minidns/dnssec/algorithms/AlgorithmMap.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2015 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.dnssec.algorithms;
+
+import de.measite.minidns.dnssec.DNSSECValidatorInitializationException;
+import de.measite.minidns.dnssec.DigestCalculator;
+import de.measite.minidns.dnssec.SignatureVerifier;
+import de.measite.minidns.record.NSEC3;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static de.measite.minidns.DNSSECConstants.DIGEST_ALGORITHM_SHA1;
+import static de.measite.minidns.DNSSECConstants.DIGEST_ALGORITHM_SHA256;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_DSA;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_DSA_NSEC3_SHA1;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_ECC_GOST;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_ECDSAP256SHA256;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_ECDSAP384SHA384;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSAMD5;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSASHA1;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSASHA1_NSEC3_SHA1;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSASHA256;
+import static de.measite.minidns.DNSSECConstants.SIGNATURE_ALGORITHM_RSASHA512;
+
+public class AlgorithmMap {
+    private Logger LOGGER = Logger.getLogger(AlgorithmMap.class.getName());
+
+    private Map<Byte, DigestCalculator> dsDigestMap;
+    private Map<Byte, SignatureVerifier> signatureMap;
+    private Map<Byte, DigestCalculator> nsecDigestMap;
+
+    @SuppressWarnings("deprecation")
+    public AlgorithmMap() {
```

Comment:
I'm not sure if this Map should be static like this and if optional/recommended algorithms should be trusted out-of-the-box.

P-256 was recently deprecated as part of NSA Suite B, see e.g
https://threatpost.com/nsas-divorce-from-ecc-causing-crypto-hand-wringing/115150/

There has also been a long history of optional-but-enabled biting back. I think there should at least be a way to configure the map. (See TLS and the rollercoaster of Algorithms to enable/disable)